src/bindings/python/flux/job/submit.py

```diff
@@ -25,6 +25,10 @@ def get_id(self):
         """Return the job ID represented by this future."""
         return submit_get_id(self)
 
+    def get(self):
+        """Return the job represented by this future."""
+        super().get()
+        return submit_get_id(self)
```

Comment:
Can you explain why? The flux async won't work without it - it was completely necessary / required for it.

Reply:
Even if it was necessary, it could go in a prior commit, right? But I don't see how it is necessary. Anywhere you call the new `f.get()` implementation you could have called `submit_get_id(f)` instead.

Reply:
Yeah we could have - I wrote this with @trws and I think we liked adding the get() function. But if you absolutely hate it I guess we could remove it and call those two things separately? It would still be kind of wonky because the submit_get_id needs a self, so I like this design better.

Reply:
I like the design too, I was just asking you to put it in a separate commit. It sounds like you want to do the git history rewriting later though, which is OK.

Reply:
Isn't the `super().get()` implied by `submit_get_id()`? Someone reading the code here could get confused, so at least a comment would be helpful.

 BTW, I think the Python here was just following the C API, which needs a separate `flux_*_get_something()` to pull correct types out of a `flux_future_t` since there's no subclassing in C. I think a lot of other Flux Python APIs probably follow this same idiom. If we override `.get()` in the `SubmitFuture` class, perhaps we should do it the other subclasses of Future as well?

Reply:
(not a strong opinion here, just bringing up the question)

Reply:
Could you tell me exactly which classes you want it added to?

Reply:
We added it because we need a way to implement `.result()`.  I was actually surprised when I realized we didn't have a unified `get` method, that's what `get` was supposed to do, but that's why this is here.  Without it the `.result()` method would have to become some wretched dystopian horror of `isinstanceof` crossed with custom method names.  I was hoping we could avoid "result as implemented by Heironymus Bosch," so *all* futures must implement a common getter method, so we added `.get`.

As to the specific issue, technically yes it could be moved to a separate commit, but it was part of the implementation of result, so I'm not sure it's worth it unless the goal is to add a note to the commit message.

Now I really want to try "implementation of result getter method by Heironymus Bosch" in Stable Diffusion...